### ISSUE TITLE
"cannot run C compiled programs" 10.12.4 / Xcode 8.3 build error

### ISSUE THREAD

**Issue by iotashan** [erikkaashoek/Comskip#62]
Fresh clone, tried to build:

```
$ ./autogen.sh --verbose
Verbose output enabled
Found a configure template: ./configure.ac
Using PROJECT environment variable override: React
Preparing the React build system...please wait

Checking autoreconf version: autoreconf --version
Checking autoconf version: autoconf --version
Found GNU Autoconf version 2.69
Checking if 2.69.0 is greater than 2.52.0
Checking automake version: automake --version
Found GNU Automake version 1.15
Checking if 1.15.0 is greater than 1.6.0
Checking libtoolize version: libtoolize --version
Found GNU Libtool version 2.4.6
Checking if 2.4.6 is greater than 1.4.2
Checking aclocal version: aclocal --version
Checking autoheader version: autoheader --version
Checking whether to only output version information
Found an autom4te.cache directory, deleting it
rm -rf autom4te.cache
Detected non-GNU marker "foreign" in configure.ac
mv -f "./config.guess" "./config.guess.backup"
mv -f "./config.sub" "./config.sub.backup"

Automatically preparing build ... autoreconf  -i -f
configure.ac:10: installing './config.guess'
configure.ac:10: installing './config.sub'
rm -f "./config.guess.backup"
rm -f "./config.sub.backup"
done

The React build system is now prepared.  To build here, run:
  ./configure
  make
rm -f "./config.guess.backup"
rm -f "./config.sub.backup"

$ ./configure
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... ./install-sh -c -d
checking for gawk... no
checking for mawk... no
checking for nawk... no
checking for awk... awk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... configure: error: in `/Users/shan/dev/iotashan/Comskip':
configure: error: cannot run C compiled programs.
If you meant to cross compile, use `--host'.
See `config.log' for more details
```

```
$ brew config
HOMEBREW_VERSION: 1.1.12
ORIGIN: https://github.com/Homebrew/brew
HEAD: 8ea778f7c7ea790694485030d105a92a9416ef33
Last commit: 2 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 490b0aad825f06a4701b1cfb5b6ab8b0933bc854
Core tap last commit: 15 hours ago
HOMEBREW_PREFIX: /usr/local
HOMEBREW_REPOSITORY: /usr/local/Homebrew
HOMEBREW_CELLAR: /usr/local/Cellar
HOMEBREW_BOTTLE_DOMAIN: https://homebrew.bintray.com
CPU: octa-core 64-bit skylake
Homebrew Ruby: 2.0.0-p648
Clang: 8.1 build 802
Git: 2.11.0 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Perl: /usr/bin/perl
Python: /usr/local/bin/python => /usr/local/Cellar/python/2.7.13/Frameworks/Python.framework/Versions/2.7/bin/python2.7
Ruby: /usr/bin/ruby => /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/bin/ruby
Java: 1.8.0_112
macOS: 10.12.4-x86_64
Xcode: 8.3
CLT: 8.3.0.0.1.1490382677
X11: N/A
MacPorts/Fink: /opt/local/bin/port
```

**Comment by iotashan** [comment]
Also I did `xcode-select --install` and also agreed to the terms/conditions, still have the issue.

**Comment by tmm1** [comment]
You need to look at the config.log and see what the error is

**Comment by iotashan** [comment]
```
$ cat config.log
This file contains any messages produced by compilers while
running configure, to aid debugging if configure makes a mistake.

It was created by Comskip configure 0.81.089, which was
generated by GNU Autoconf 2.69.  Invocation command line was

  $ ./configure 

## --------- ##
## Platform. ##
## --------- ##

hostname = Shans-MacBook-Pro-Touchbar.local
uname -m = x86_64
uname -r = 16.5.0
uname -s = Darwin
uname -v = Darwin Kernel Version 16.5.0: Fri Mar  3 16:52:33 PST 2017; root:xnu-3789.51.2~3/RELEASE_X86_64

/usr/bin/uname -p = i386
/bin/uname -X     = unknown

/bin/arch              = unknown
/usr/bin/arch -k       = unknown
/usr/convex/getsysinfo = unknown
/usr/bin/hostinfo      = Mach kernel version:
	 Darwin Kernel Version 16.5.0: Fri Mar  3 16:52:33 PST 2017; root:xnu-3789.51.2~3/RELEASE_X86_64
Kernel configured for up to 8 processors.
4 processors are physically available.
8 processors are logically available.
Processor type: x86_64h (Intel x86-64h Haswell)
Processors active: 0 1 2 3 4 5 6 7
Primary memory available: 16.00 gigabytes
Default processor set: 439 tasks, 2614 threads, 8 processors
Load average: 6.26, Mach factor: 3.38
/bin/machine           = unknown
/usr/bin/oslevel       = unknown
/bin/universe          = unknown

PATH: /Users/shan/.nvm/versions/node/v7.4.0/bin
PATH: /usr/local/bin
PATH: /usr/bin
PATH: /bin
PATH: /usr/sbin
PATH: /sbin


## ----------- ##
## Core tests. ##
## ----------- ##

configure:1937: checking for a BSD-compatible install
configure:2005: result: /usr/bin/install -c
configure:2016: checking whether build environment is sane
configure:2071: result: yes
configure:2222: checking for a thread-safe mkdir -p
configure:2261: result: ./install-sh -c -d
configure:2268: checking for gawk
configure:2298: result: no
configure:2268: checking for mawk
configure:2298: result: no
configure:2268: checking for nawk
configure:2298: result: no
configure:2268: checking for awk
configure:2284: found /usr/bin/awk
configure:2295: result: awk
configure:2306: checking whether make sets $(MAKE)
configure:2328: result: yes
configure:2357: checking whether make supports nested variables
configure:2374: result: yes
configure:2550: checking for gcc
configure:2566: found /usr/bin/gcc
configure:2577: result: gcc
configure:2806: checking for C compiler version
configure:2815: gcc --version >&5
Apple LLVM version 8.1.0 (clang-802.0.38)
Target: x86_64-apple-darwin16.5.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
Configured with: --prefix=/Applications/Xcode.app/Contents/Developer/usr --with-gxx-include-dir=/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator10.3.sdk/usr/include/c++/4.2.1
configure:2826: $? = 0
configure:2815: gcc -v >&5
Configured with: --prefix=/Applications/Xcode.app/Contents/Developer/usr --with-gxx-include-dir=/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator10.3.sdk/usr/include/c++/4.2.1
Apple LLVM version 8.1.0 (clang-802.0.38)
Target: x86_64-apple-darwin16.5.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
configure:2826: $? = 0
configure:2815: gcc -V >&5
clang: error: argument to '-V' is missing (expected 1 value)
clang: error: no input files
configure:2826: $? = 1
configure:2815: gcc -qversion >&5
clang: error: unknown argument: '-qversion'
clang: error: no input files
configure:2826: $? = 1
configure:2846: checking whether the C compiler works
configure:2868: gcc    conftest.c  >&5
configure:2872: $? = 0
configure:2920: result: yes
configure:2923: checking for C compiler default output file name
configure:2925: result: a.out
configure:2931: checking for suffix of executables
configure:2938: gcc -o conftest    conftest.c  >&5
configure:2942: $? = 0
configure:2964: result: 
configure:2986: checking whether we are cross compiling
configure:2994: gcc -o conftest    conftest.c  >&5
configure:2998: $? = 0
configure:3005: ./conftest
dyld: mach-o, but built for simulator (not macOS)
./configure: line 3007: 60627 Abort trap: 6           ./conftest$ac_cv_exeext
configure:3009: $? = 134
configure:3016: error: in `/Users/shan/dev/iotashan/Comskip':
configure:3018: error: cannot run C compiled programs.
If you meant to cross compile, use `--host'.
See `config.log' for more details

## ---------------- ##
## Cache variables. ##
## ---------------- ##

ac_cv_env_CC_set=
ac_cv_env_CC_value=
ac_cv_env_CFLAGS_set=
ac_cv_env_CFLAGS_value=
ac_cv_env_CPPFLAGS_set=
ac_cv_env_CPPFLAGS_value=
ac_cv_env_LDFLAGS_set=
ac_cv_env_LDFLAGS_value=
ac_cv_env_LIBS_set=
ac_cv_env_LIBS_value=
ac_cv_env_PKG_CONFIG_LIBDIR_set=
ac_cv_env_PKG_CONFIG_LIBDIR_value=
ac_cv_env_PKG_CONFIG_PATH_set=
ac_cv_env_PKG_CONFIG_PATH_value=
ac_cv_env_PKG_CONFIG_set=
ac_cv_env_PKG_CONFIG_value=
ac_cv_env_argtable2_CFLAGS_set=
ac_cv_env_argtable2_CFLAGS_value=
ac_cv_env_argtable2_LIBS_set=
ac_cv_env_argtable2_LIBS_value=
ac_cv_env_build_alias_set=
ac_cv_env_build_alias_value=
ac_cv_env_ffmpeg_CFLAGS_set=
ac_cv_env_ffmpeg_CFLAGS_value=
ac_cv_env_ffmpeg_LIBS_set=
ac_cv_env_ffmpeg_LIBS_value=
ac_cv_env_host_alias_set=
ac_cv_env_host_alias_value=
ac_cv_env_sdl_CFLAGS_set=
ac_cv_env_sdl_CFLAGS_value=
ac_cv_env_sdl_LIBS_set=
ac_cv_env_sdl_LIBS_value=
ac_cv_env_target_alias_set=
ac_cv_env_target_alias_value=
ac_cv_path_install='/usr/bin/install -c'
ac_cv_prog_AWK=awk
ac_cv_prog_ac_ct_CC=gcc
ac_cv_prog_make_make_set=yes
am_cv_make_support_nested_variables=yes

## ----------------- ##
## Output variables. ##
## ----------------- ##

ACLOCAL='${SHELL} /Users/shan/dev/iotashan/Comskip/missing aclocal-1.15'
AMDEPBACKSLASH=''
AMDEP_FALSE=''
AMDEP_TRUE=''
AMTAR='$${TAR-tar}'
AM_BACKSLASH='\'
AM_DEFAULT_V='$(AM_DEFAULT_VERBOSITY)'
AM_DEFAULT_VERBOSITY='1'
AM_V='$(V)'
AUTOCONF='${SHELL} /Users/shan/dev/iotashan/Comskip/missing autoconf'
AUTOHEADER='${SHELL} /Users/shan/dev/iotashan/Comskip/missing autoheader'
AUTOMAKE='${SHELL} /Users/shan/dev/iotashan/Comskip/missing automake-1.15'
AWK='awk'
BUILD_GUI_FALSE=''
BUILD_GUI_TRUE=''
CC='gcc'
CCDEPMODE=''
CFLAGS=''
CPPFLAGS=''
CYGPATH_W='echo'
DEFS=''
DEPDIR=''
ECHO_C='\c'
ECHO_N=''
ECHO_T=''
ENABLE_DEBUG_FALSE=''
ENABLE_DEBUG_TRUE=''
ENABLE_DONATOR_FALSE=''
ENABLE_DONATOR_TRUE=''
ENABLE_STATIC_FALSE=''
ENABLE_STATIC_TRUE=''
EXEEXT=''
INSTALL_DATA='${INSTALL} -m 644'
INSTALL_PROGRAM='${INSTALL}'
INSTALL_SCRIPT='${INSTALL}'
INSTALL_STRIP_PROGRAM='$(install_sh) -c -s'
LDFLAGS=''
LIBOBJS=''
LIBS=''
LTLIBOBJS=''
MAKEINFO='${SHELL} /Users/shan/dev/iotashan/Comskip/missing makeinfo'
MKDIR_P='./install-sh -c -d'
OBJEXT=''
PACKAGE='comskip'
PACKAGE_BUGREPORT='https://github.com/erikkaashoek/Comskip/issues'
PACKAGE_NAME='Comskip'
PACKAGE_STRING='Comskip 0.81.089'
PACKAGE_TARNAME='comskip'
PACKAGE_URL=''
PACKAGE_VERSION='0.81.089'
PATH_SEPARATOR=':'
PKG_CONFIG=''
PKG_CONFIG_LIBDIR=''
PKG_CONFIG_PATH=''
PLATFORM_WIN32_FALSE=''
PLATFORM_WIN32_TRUE=''
SET_MAKE=''
SHELL='/bin/sh'
STRIP=''
VERSION='0.81.089'
ac_ct_CC='gcc'
am__EXEEXT_FALSE=''
am__EXEEXT_TRUE=''
am__fastdepCC_FALSE=''
am__fastdepCC_TRUE=''
am__include=''
am__isrc=''
am__leading_dot='.'
am__nodep=''
am__quote=''
am__tar='$${TAR-tar} chof - "$$tardir"'
am__untar='$${TAR-tar} xf -'
argtable2_CFLAGS=''
argtable2_LIBS=''
bindir='${exec_prefix}/bin'
build=''
build_alias=''
build_cpu=''
build_os=''
build_vendor=''
datadir='${datarootdir}'
datarootdir='${prefix}/share'
docdir='${datarootdir}/doc/${PACKAGE_TARNAME}'
dvidir='${docdir}'
exec_prefix='NONE'
ffmpeg_CFLAGS=''
ffmpeg_LIBS=''
host=''
host_alias=''
host_cpu=''
host_os=''
host_vendor=''
htmldir='${docdir}'
includedir='${prefix}/include'
infodir='${datarootdir}/info'
install_sh='${SHELL} /Users/shan/dev/iotashan/Comskip/install-sh'
libdir='${exec_prefix}/lib'
libexecdir='${exec_prefix}/libexec'
localedir='${datarootdir}/locale'
localstatedir='${prefix}/var'
mandir='${datarootdir}/man'
mkdir_p='$(MKDIR_P)'
oldincludedir='/usr/include'
pdfdir='${docdir}'
prefix='NONE'
program_transform_name='s,x,x,'
psdir='${docdir}'
sbindir='${exec_prefix}/sbin'
sdl_CFLAGS=''
sdl_LIBS=''
sharedstatedir='${prefix}/com'
sysconfdir='${prefix}/etc'
target_alias=''

## ----------- ##
## confdefs.h. ##
## ----------- ##

/* confdefs.h */
#define PACKAGE_NAME "Comskip"
#define PACKAGE_TARNAME "comskip"
#define PACKAGE_VERSION "0.81.089"
#define PACKAGE_STRING "Comskip 0.81.089"
#define PACKAGE_BUGREPORT "https://github.com/erikkaashoek/Comskip/issues"
#define PACKAGE_URL ""
#define PACKAGE "comskip"
#define VERSION "0.81.089"

configure: exit 1
```

**Comment by tmm1** [comment]
Something is wrong with your Xcode 

>configure:3005: ./conftest
dyld: mach-o, but built for simulator (not macOS)
>./configure: line 3007: 60627 Abort trap: 6
